crates/sui-graphql-rpc/src/types/epoch.rs

```diff
@@ -292,43 +290,35 @@ impl Epoch {
     /// cursor).
     pub(crate) async fn query_latest_at(
         db: &Db,
-        checkpoint_viewed_at: Option<u64>,
+        checkpoint_viewed_at: u64,
     ) -> Result<Option<Self>, Error> {
         use epochs::dsl;
 
-        let (stored, checkpoint_viewed_at): (Option<QueryableEpochInfo>, u64) = db
-            .execute_repeatable(move |conn| {
-                let checkpoint_viewed_at = match checkpoint_viewed_at {
-                    Some(value) => Ok(value),
-                    None => Checkpoint::available_range(conn).map(|(_, rhs)| rhs),
-                }?;
-
-                let stored = conn
-                    .first(move || {
-                        let mut query = dsl::epochs
-                            .select(QueryableEpochInfo::as_select())
-                            .order_by(dsl::epoch.desc())
-                            .into_boxed();
-
-                        // Bound the query on `checkpoint_viewed_at` by filtering for the epoch
-                        // whose `first_checkpoint_id <= checkpoint_viewed_at`, selecting the epoch
-                        // with the largest `first_checkpoint_id` among the filtered set.
-                        query = query
-                            .filter(dsl::first_checkpoint_id.le(checkpoint_viewed_at as i64))
-                            .order_by(dsl::first_checkpoint_id.desc());
-
-                        query
-                    })
-                    .optional()?;
-
-                Ok::<_, diesel::result::Error>((stored, checkpoint_viewed_at))
+        let stored: Option<QueryableEpochInfo> = db
+            .execute(move |conn| {
+                conn.first(move || {
+                    let mut query = dsl::epochs
+                        .select(QueryableEpochInfo::as_select())
+                        .order_by(dsl::epoch.desc())
+                        .into_boxed();
+
+                    // Bound the query on `checkpoint_viewed_at` by filtering for the epoch
+                    // whose `first_checkpoint_id <= checkpoint_viewed_at`, selecting the epoch
+                    // with the largest `first_checkpoint_id` among the filtered set.
+                    query = query
+                        .filter(dsl::first_checkpoint_id.le(checkpoint_viewed_at as i64))
+                        .order_by(dsl::first_checkpoint_id.desc());
```

Comment:
Do we still need the first order by epoch? I think ordering by first checkpoint id is sufficient since this should be 1 to 1

Reply:
Yes, I think you're right!